libpod/healthcheck_linux.go

```diff
@@ -53,6 +57,12 @@ func (c *Container) createTimer(interval string, isStartup bool) error {
 	if output, err := systemdRun.CombinedOutput(); err != nil {
 		return fmt.Errorf("%s", output)
 	}
+
+	c.state.HCUnitName = hcUnitName
+	if err := c.save(); err != nil {
+		return fmt.Errorf("saving container %s healthcheck unit name: %w", c.ID(), err)
+	}
```

Comment:
should we let the caller do this? At the risk of some misuse but looking at the caller in init() it does a save directly before calling this so I think it would make sense to move this after it only save once there for perfomance reasons.

Reply:
This can be called in more than one place (e.g. transitioning from a startup healthcheck to a regular healthcheck) so I'm a bit hesitant to do that

Reply:
I think it is little risk given only two callers and it seems very unlikely to add more but I am also fine keeping, out of curiosity I just added this patch to count the calls on startup
```
diff --git a/libpod/container.go b/libpod/container.go
index 8e188b33c..046aa37ae 100644
--- a/libpod/container.go
+++ b/libpod/container.go
@@ -129,6 +129,8 @@ type Container struct {
 
        slirp4netnsSubnet *net.IPNet
        pastaResult       *pasta.SetupResult
+
+       saveCounter int
 }
 
 // ContainerState contains the current state of the container
diff --git a/libpod/container_internal.go b/libpod/container_internal.go
index ffb060b92..67fe560fa 100644
--- a/libpod/container_internal.go
+++ b/libpod/container_internal.go
@@ -817,6 +817,8 @@ func (c *Container) save() error {
        if err := c.runtime.state.SaveContainer(c); err != nil {
                return fmt.Errorf("saving container %s state: %w", c.ID(), err)
        }
+       c.saveCounter++
+       fmt.Println(c.saveCounter)
        return nil
 }
```

The answer for a simple image with default options only *3* which seems very good already so one more is likely not noticeable and safer